File: collatz.lua
num = 1
counterMax = 0
while true do
    calc = num
    counter = 1
    while calc ~= 1 do
        if calc % 2 == 0 then
            calc = tonumber(calc) / 2
        else
            calc = 3 * tonumber(calc) + 1
        counter = counter + 1
        end
    end
    
    if counterMax < counter then
        counterMax = counter
        print('Lua Collatz from '..tostring(num).." of len "..tostring(counter))
    end
    num = num + 1
end

File: collatz.ts
let num = 1
let counterMax = 0
while (true) {
    let calc = num
    let counter = 1
    while (calc != 1) {
        if (calc % 2 == 0) {
            calc = calc / 2
        } else {
            calc = 3 * calc + 1
        }
        counter++
    }
    if (counterMax < counter) {
        counterMax = counter
        console.log(`JS Collatz from ${num} of len ${counter}`)
    }
    num++
}